doc/topics/the-event-loop-timers-and-nexttick

```diff
@@ -0,0 +1,174 @@
+is concerned, its happening **synchronously**, but as far as the user is
+concerned, it is happening **asynchronously** because the API of apiCall() was
+written to always be asynchronous.
+
+This philosophy can lead to some potentially problematic situations.  Take this
+snippet for example:
+
+```javascript
+// this has an asynchronous signature, but calls callback synchronously
+function someAsyncApiCall (callback) {
+  callback();
+};
+
+// the callback is called before `someAsyncApiCall` completes.
+someAsyncApiCall(function () {
+
+  // since someAsyncApiCall has completed, bar hasn't been assigned any value
+  console.log('bar', bar); // undefined
+
+});
+
+var bar = 1;
+```
+
+The user defines `someAsyncApiCall()` to have an asynchronous signature,
+actually operates synchronously.  When it is called, the callback provided to
+`someAsyncApiCall ()` is called in the same phase of the event loop
+because `someAsyncApiCall()` doesn't actually do anything asynchronously.  As a
+result, the callback tries to reference `bar` but it may not have that variable
+in scope yet because the script has not been able to run to completion.
+
+By placing it in a nextTick, the script
+still has the ability to run to completion, allowing all the variables,
+functions, etc., to be initialized prior to the callback being called.   It also
+has the advantage of not allowing the event loop to continue.  It may be useful
+that the user be alerted to an error before the event loop is allowed to
+continue.
+
+## process.nextTick() vs setImmediate()
+
+We have two calls that are similar as far as users are concerned, but their
+names are confusing.
+
+* nextTick fires immediately on the same phase
+* setImmediate fires on the following iteration or 'tick' of the event loop
+
+In essence, the names should be swapped.  nextTick fires more immediately than
+setImmediate but this is an artifact of the past which is unlikely to change.
+Making this switch would break a large percentage of the packages on npm.
+Every day more new modules are being added, which mean every day we wait, more
+potential breakages occur.  While they are confusing, the names themselves won't change.
+
+*We recommend developers use setImmediate in all cases because its easier to
+reason about.*
+
+## Two reasons to use nextTick:
+
+1. Allow users to handle errors, cleanup any then unneeded resources, or
+perhaps try the request again before the event loop continues.
+
+2. If you were to run a function constructor that was to, say, inherit from
+`EventEmitter` and it wanted to call an event within the constructor.  You can't
+emit an event from the constructor immediately because the script will not have
+processed to the point where the user assigns a callback to that event.  So,
+within the constructor itself, you can set a callback to emit the event after
+the constructor has finished, which provides the expected results.
```

Comment:
This begs the questions:
* does the event loop start *once* the input script has **finished** execution **--OR--**
* does the event loop start *then* execute the input script, and if so, in which 'phase' is the input script executed? In the `poll` phase...?

If it is the former, then it seems to me that `nextTick()` would be the correct choice because it will be called once the input script has finished executing which prevents any possibility of I/O events being processed before your callback is executed, as described here: https://github.com/nodejs/docs/issues/34#issuecomment-176902809

If it is the latter and it IS in the `poll` phase, then it seems it wouldn't make much difference since the `setImmediate` phase is right after the `poll` phase.  

From what I've gathered, I believe the former provides the most accurate analogy for this example.